core/controllers/base.py

```diff
@@ -304,12 +304,7 @@ def render_json(self, values):
         counters.JSON_RESPONSE_TIME_SECS.inc(increment=processing_time)
         counters.JSON_RESPONSE_COUNT.inc()
 
-    def render_template(
-            self, filename, values=None, iframe_restriction='DENY',
-            redirect_url_on_logout=None):
-        if values is None:
-            values = self.values
-
+    def update_base_data(self, values):
```

Comment:
I think it would be cleaner if this didn't take any args, and returned a dict. The caller can update meta_name and meta_description if needed.